src/rust/src/tests/signatures.rs

```diff
@@ -481,30 +480,20 @@ fn redjubjub_signatures() {
     ];
 
     for tv in test_vectors {
-        let sk = PrivateKey::read(&tv.sk[..]).unwrap();
-        let vk = PublicKey::read(&tv.vk[..]).unwrap();
-        let rvk = PublicKey::read(&tv.rvk[..]).unwrap();
-        let sig = Signature::read(&tv.sig[..]).unwrap();
-        let rsig = Signature::read(&tv.rsig[..]).unwrap();
+        let sk = redjubjub::SigningKey::<SpendAuth>::try_from(tv.sk).unwrap();
+        let vk = redjubjub::VerificationKey::<SpendAuth>::try_from(tv.vk).unwrap();
+        let rvk = redjubjub::VerificationKey::<SpendAuth>::try_from(tv.rvk).unwrap();
+        let sig = redjubjub::Signature::<SpendAuth>::from(tv.sig);
+        let rsig = redjubjub::Signature::<SpendAuth>::from(tv.rsig);
 
         let alpha = jubjub::Scalar::from_bytes(&tv.alpha).unwrap();
 
-        {
-            let mut vec = Vec::new();
-            sk.randomize(alpha).write(&mut vec).unwrap();
-            assert_eq!(&vec, &tv.rsk);
-        }
-        {
-            let mut vec = Vec::new();
-            vk.randomize(alpha, SPENDING_KEY_GENERATOR)
-                .write(&mut vec)
-                .unwrap();
-            assert_eq!(&vec, &tv.rvk);
-        }
+        assert_eq!(<[u8; 32]>::from(sk.randomize(&alpha)), tv.rsk);
+        assert_eq!(<[u8; 32]>::from(vk.randomize(&alpha)), tv.rvk);
 
-        assert!(vk.verify(&tv.m, &sig, SPENDING_KEY_GENERATOR));
-        assert!(rvk.verify(&tv.m, &rsig, SPENDING_KEY_GENERATOR));
-        assert!(!vk.verify(&tv.m, &rsig, SPENDING_KEY_GENERATOR));
-        assert!(!rvk.verify(&tv.m, &sig, SPENDING_KEY_GENERATOR));
+        assert_eq!(vk.verify(&tv.m, &sig), Ok(()));
+        assert_eq!(rvk.verify(&tv.m, &rsig), Ok(()));
+        assert!(matches!(vk.verify(&tv.m, &rsig), Err(_)));
+        assert!(matches!(rvk.verify(&tv.m, &sig), Err(_)));
     }
```

Comment:
Something about this change from `zcash_primitives::sapling::redjubjub` (which is being removed in https://github.com/zcash/librustzcash/pull/1056) to the `redjubjub` crate is causing the test vectors to fail.

Reply:
Aha: the problem is that the test vectors were written before key prefixing was added to RedJubjub. `zcash_primitives::sapling::redjubjub` was never updated to include key prefixing, instead leaving it to the callsites. The `redjubjub` crate correctly includes key prefixing, but that means the old test vectors cannot be used with it. I'll open a PR to regenerate those test vectors.